src/WinRT.Runtime/Attributes.cs

```diff
@@ -230,6 +230,39 @@ public WinRTAssemblyExportsTypeAttribute(Type type)
         public Type Type { get; }
     }
 
+    /// <summary>
+    /// An attribute used to indicate the properties which are bindable via the <see cref="Microsoft.UI.Xaml.Data.ICustomProperty"/> implementation provided for use in WinUI scenarios.
+    /// The type which this attribute is placed on also needs to be marked partial and needs to be non-generic.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Class, Inherited = false, AllowMultiple = false)]
+#if EMBED
+    internal
+#else
+    public
+#endif
+    sealed class BindableCustomPropertyAttribute : Attribute
+    {
+        /// <summary>
+        /// Marks all public properties as bindable.
+        /// </summary>
+        public BindableCustomPropertyAttribute()
+        {
+        }
+
+        /// <summary>
+        /// Marks the specified public properties as bindable.
+        /// </summary>
+        /// <param name="propertyNames">The name of the non-indexer public properties to mark as bindable.</param>
+        /// <param name="indexerPropertyTypes">The parameter type of the indexer public properties to mark as bindable.</param>
+        public BindableCustomPropertyAttribute(string[] propertyNames, Type[] indexerPropertyTypes)
+        {
+            PropertyNames = propertyNames;
+            IndexerPropertyTypes = indexerPropertyTypes;
+        }
+
+        internal string[] PropertyNames { get; }
```

Comment:
Nit: should these two be internal? All other attribute properties are public. Not a big deal, but I think it's common practice for these kind of attribute properties just storing constructor parameters to be public? We should also add XML docs if so.

Reply:
I kept them internal to make it easier to change stuff if needed and the value of them didn't really make sense in the default constructor case where it really means all properties.